olp-cpp-sdk-core/include/olp/core/client/BackdownStrategy.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <chrono>
+#include <random>
+#include <olp/core/CoreApi.h>
+
+namespace olp {
+namespace client {
+
+/**
+ * @brief Functor which computes wait time for the next retry attempt via
+ * exponential backoff with added jitter.
+ *
+ * This backoff strategy is based on the exponential wait time approach, i.e.
+ * when wait time exponentially grows with each retry attempt, but with added
+ * randomization.
+ *
+ * The actual formula can be described as:
+ * @code{.unparsed}
+ * wait_time = random_between(0, initial_backdown_period_msec * (2 ^
+ * retry_count))
+ * @endcode
+ */
+struct CORE_API ExponentialBackdownStrategy {
+ public:
+  /**
+   * @brief Function to compute next retry attempt wait time based on number of
+   * retries and initial backdown period.
+   *
+   * @param initial_backdown_period_msec Initial backdown period in
+   * milliseconds.
+   * @param retry_count Number of retries already made.
+   *
+   * @return Wait time in milliseconds for the next retry attempt.
+   */
+  std::chrono::milliseconds operator()(
+      std::chrono::milliseconds initial_backdown_period, size_t retry_count) {
+    const auto exponential_wait_time =
+        initial_backdown_period.count() * (1 << retry_count);
+
+    static thread_local std::mt19937 kGenerator(std::random_device{}());
```

Comment:
Just asking. Would it make sense to set the seed?

Reply:
Actually, I'm seeding the Mersenne Twister RNG with a seed value generated by `std::random_device`.  Note that random_device is a bit heavy-weight, as it can access OS resources (like reading https://en.wikipedia.org/wiki//dev/random file). 

Alternatively, we can use light-weighter approach, like:
-  seeding some magic seed value like 42;
- good'n'old `time(nullptr)`; 
- default seed value of Mersenne Twister (`std::mt19937::default_seed`);
- 'seed' a seed sequence (https://en.cppreference.com/w/cpp/numeric/random/seed_seq) for random generator, if we want more entropy, but it looks more like an over-engineering to me in such simple task as calculate next backdown strategy.

However, I have chosen the random device approach, as it is de-facto very popular among other examples that I saw on SO/C++ community regarding usage of C++ pRNG. 

P.S.: Also, after a bit of research on this topic, we can use a C++11-ish `srand()` approach - http://www.cplusplus.com/reference/random/minstd_rand/ instead Mersenne Twister approach, if we want a really simple pRNG. However, MT is more like default random numbers generator and should be used by default.